### PR TITLE
Kubernetes pods should be able to communicate with kubernetes services.

### PR DESCRIPTION
**Pull request by RostyslavFridman**
Add route to kubernetes services through management interface.

### CODE REVIEW

File: bin/ovn-k8s-overlay [hunk 1]
```diff
@@ -162,12 +162,16 @@ def create_management_port(node_name, local_subnet, cluster_subnet):
               "--", "set", "logical_switch", node_name,
               "other-config:subnet=" + local_subnet)
 
+    # Create route to kubernetes services
+    ovn_nbctl("--may-exist", "lr-route-add", node_name,
```
Comment (salv-orlando) [inline bin/ovn-k8s-overlay]: create_management_port is called both by master_init and minion_init - in the latter case node_name will be the name of the "minion" node - and there won't be a OVN router with that name, I think

File: bin/ovn-k8s-overlay [hunk 2]
```diff
@@ -230,14 +234,14 @@ def master_init(args):
...
-                           args.cluster_ip_subnet)
+                           args.cluster_ip_subnet, args.service_ip_subnet)
 
 
 def minion_init(args):
     fetch_ovn_nb()
 
     if not args.node_name or not args.minion_switch_subnet \
-       or not args.cluster_ip_subnet:
+       or not args.cluster_ip_subnet or not args.service_ip_subnet:
```
Comment (salv-orlando) [inline bin/ovn-k8s-overlay]: Perhaps it might be sufficient to add the route for kubernetes service during master_init only?

File: bin/ovn-k8s-overlay [hunk 3]
```diff
@@ -416,33 +421,35 @@ def main():
...
+    parser_minion_init.add_argument('--service-ip-subnet', required=True,
+                                    help="Subnet for kubernetes services.")
     parser_minion_init.add_argument('--node-name',
                                     required=True,
                                     help="A unique node name.")
     parser_minion_init.set_defaults(func=minion_init)
 
     # Parser for sub-command 'gateway-init'.
-    parser_minion_init = subparsers.add_parser(
+    parser_gateway_init = subparsers.add_parser(
```
Comment (salv-orlando) [inline bin/ovn-k8s-overlay]: thanks for spotting this bad reuse of a local variable.
Personally, I would do a separate PR for this, but I do not have a strong opinion on this regard.

**Comment by shettyg** [comment]
I don't understand what this patch is trying to do. Specifically I do not understand what KUBERNETES_SERVICES_SUBNET here? Is this the subnet that k8s assigns virtual ip addresses from when we create a k8s service? If so, whenever a k8s service is created, the watcher will create load balancer entries and if the pod tries to access it, it will work fine. (This is assuming that you have the Open vSwitch kernel module from the OVS repo. If it is from upstream linux, you need a minimum version of 4.6. This is mainly because you need NAT support.)

**Comment by salv-orlando** [comment]
I was confused by intentions too. Perhaps the commit message could be improved by adding more information.

for instance I thought this patch was about ensuring a pod could contact kubernetes core services such as kubelet, scheduler, etc.

**Comment by RostyslavFridman** [comment]
Sorry, I suppose it was some issue on my side that prevented pods to contact services without the additional route. I've recreated my testbed and it is working now.
I'm closing this request as it is clearly wrong.
I will send separate PR with variable fix.
